src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbKey.java

```diff
@@ -0,0 +1,52 @@
+package com.exasol.adapter.dynamodb.dynamodbmetadata;
+
+import java.io.Serializable;
+
+import com.exasol.adapter.dynamodb.documentpath.DocumentPathExpression;
+
+/**
+ * This class represents a DynamoDB key. It can either be a primary key or a secondary index.
+ */
+public class DynamodbKey implements Serializable {
+    private static final long serialVersionUID = 4048033058983909214L;
+    private final DocumentPathExpression partitionKey;
+    private final DocumentPathExpression sortKey;
+
+    /**
+     * Creates an instance of {@link DynamodbKey}.
+     * 
+     * @param partitionKey partition key of this key
+     * @param sortKey      sort key of this key. {@code null} if no sort key is present.
+     */
+    public DynamodbKey(final DocumentPathExpression partitionKey, final DocumentPathExpression sortKey) {
```

Comment:
Maybe having two constructors is better than sending null to this one

Reply:
I use this constructor only in `DynamodbTableMetadataFactory`. There it is more practicable to do it with the optional here. I kept it public for the use in other packages tests